pwa/app/components/tba/matchResultsTable.tsx

```diff
@@ -0,0 +1,241 @@
+import { Link } from '@remix-run/react';
+import { type VariantProps, cva } from 'class-variance-authority';
+import type React from 'react';
+import { Fragment } from 'react';
+import { Match } from '~/api/v3';
+import {
+  cn,
+  sortMatchComparator,
+  timestampsAreOnDifferentDays,
+  zip,
+} from '~/lib/utils';
+import PlayCircle from '~icons/bi/play-circle';
+
+const cellVariants = cva('', {
+  variants: {
+    matchResult: {
+      winner: 'font-semibold',
+      loser: '',
+    },
+    allianceColor: {
+      red: 'bg-alliance-red-light',
+      blue: 'bg-alliance-blue-light',
+    },
+    teamOrScore: {
+      team: '',
+      score: '',
+    },
+  },
+  defaultVariants: {
+    matchResult: 'loser',
+    allianceColor: undefined,
+    teamOrScore: 'team',
+  },
+  compoundVariants: [
+    {
+      allianceColor: 'red',
+      teamOrScore: 'score',
+      class: 'bg-alliance-red-dark',
+    },
+    {
+      allianceColor: 'blue',
+      teamOrScore: 'score',
+      class: 'bg-alliance-blue-dark',
+    },
+  ],
+});
+
+interface CellProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof cellVariants> {
+  dq?: boolean;
+  surrogate?: boolean;
+}
+function GridCell({
+  className,
+  matchResult,
+  allianceColor,
+  teamOrScore,
+  dq,
+  surrogate,
+  ...props
+}: CellProps) {
+  return (
+    <div
+      className={cn(
+        cellVariants({ matchResult, allianceColor, teamOrScore }),
+        className,
+        {
+          'line-through': dq,
+          'underline decoration-dotted': surrogate,
+        },
+      )}
+      {...props}
+    />
+  );
+}
+
+// TODO: Add support for single elim brackets
+function matchTitle(match: Match): string {
+  if (match.comp_level === 'f') {
+    return `Finals ${match.match_number}`;
+  }
+
+  if (match.comp_level === 'qm') {
+    return `Quals ${match.match_number}`;
+  }
+
+  return `Match ${match.set_number}`;
+}
+
+function maybeGetFirstMatchVideoURL(match: Match): string | undefined {
+  if (match.videos === undefined || match.videos.length === 0) {
+    return undefined;
+  }
+
+  return `https://www.youtube.com/watch?v=${match.videos[0].key}`;
+}
+
+// tailwindgen.com is recommended if you're editing the grid classes
+export default function MatchResultsTable(props: {
```

Comment:
Can we destructure `props` here? i.e. `MatchResultsTable({matches, title}:{ matches: Match[], title: string})`